packages/aws-cdk-lib/aws-rds/test/cluster.test.ts

```diff
@@ -1798,15 +1819,139 @@ describe('cluster', () => {
         new DatabaseCluster(stack, 'Database', {
           engine: DatabaseClusterEngine.AURORA,
           vpc,
-          writer: ClusterInstance.provisioned('writer', {
+          writer: ClusterInstance.provisioned('writer'),
+          enablePerformanceInsights: false,
+          performanceInsightRetention: PerformanceInsightRetention.DEFAULT,
+        });
+      }).toThrow(/`enablePerformanceInsights` disabled, but `performanceInsightRetention` or `performanceInsightEncryptionKey` was set/);
+    });
+
+    test('warn if performance insights is enabled at cluster level but disabled on writer instance', () => {
+      // GIVEN
+      const stack = testStack();
+      const vpc = new ec2.Vpc(stack, 'VPC');
+
+      // WHEN
+      new DatabaseCluster(stack, 'Database', {
+        engine: DatabaseClusterEngine.AURORA,
+        vpc,
+        writer: ClusterInstance.provisioned('writer', {
+          enablePerformanceInsights: false,
+        }),
+        enablePerformanceInsights: true,
+      });
+
+      // THEN
+      Annotations.fromStack(stack).hasWarning('*',
+        'Performance Insights is enabled on cluster \'Database\' at cluster level, but disabled for instance \'writer\'. '+
+        'However, Performance Insights for this instance will also be automatically enabled if enabled at cluster level. [ack: @aws-cdk/aws-rds:instancePerformanceInsightsOverridden]',
+      );
+    });
+
+    test('warn if performance insights is enabled at cluster level but disabled on reader instance', () => {
```

Comment:
We could remove this as redundant with `warn if performance insights is enabled at cluster level but disabled on writer and reader instances`

Reply:
If we turn them off, we will not be able to see whether the validation process has been done for both writer and reader. Therefore, may I suggest that we turn the test (`warn if performance insights is enabled at cluster level but disabled on writer and reader instances`) off?

Reply:
Sorry, I misunderstood. This is about the warning, so we could see it. I'll take it.

Reply:
done: https://github.com/aws/aws-cdk/pull/31385/commits/a167d1f12169f219eca03b811b50e70b7563566a